python/foglamp/common/configuration_manager.py

```diff
@@ -210,17 +210,44 @@ def get_entry_val(k):
         return category_info
 
     async def _read_all_groups(self, root):
-        payload = PayloadBuilder().SELECT("key", "description", "value").chain_payload()
-        if root is True:
-            payload = PayloadBuilder(payload).WHERE(["value", "=", "{}"]).payload()
+        group_info = []
+        if root:
+            # SELECT DISTINCT parent FROM category_children WHERE parent NOT IN (SELECT child FROM category_children);
```

Comment:
Please see latest comment in Jira, it is the list of categories that is the base list to remove entries from, as you may have a root category that has no children and hence is not a parent.